### DISCUSSION TITLE
Refresh Index for Vault needs a Rebuild Index option

### DISCUSSION THREAD

**Discussion by RobVess** [logancyang/obsidian-copilot#557] [Q&A]
- [ ] Screenshot of note + Copilot chat pane + dev console added **(required)**
![image](https://github.com/logancyang/obsidian-copilot/assets/9983337/4cca8fef-2622-44c1-b60e-cecc59582fa8)
2 out of the 3 files listed here don't exist or have been renamed (1 each)

**Describe how to reproduce**
Add in a range of Markdown documents to a Vault.
Press Reindex, it will add items to the index
Rename a file. Press Refresh index. The old file is still there as a source.
Delete a file. Press Refresh on the index. Old file is still in the index as a source.

**Expected behavior**
Provide a "Rebuild Index" button which clears out the entire vault index and reindexes from scratch to clear out issues.

**Comment by logancyang** [comment]
Does "force re-index" not work? If it doesn't let me know.

**Comment by RobVess** [comment]
The reindex button appears to just do the change set (last 2 or 3 docs) not the whole set and appears to miss renamed files

Rob.
________________________________
